test/extensions/filters/http/ratelimit/ratelimit_headers_test.cc

```diff
@@ -0,0 +1,101 @@
+#include <string>
+#include <vector>
+
+#include "extensions/filters/http/ratelimit/ratelimit_headers.h"
+
+#include "test/extensions/filters/common/ratelimit/utils.h"
+#include "test/mocks/http/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace RateLimitFilter {
+namespace {
+
+using Envoy::RateLimit::buildDescriptorStatus;
+using Filters::Common::RateLimit::DescriptorStatusList;
+
+struct RateLimitHeadersTestCase {
+  Http::TestResponseHeaderMapImpl expected_headers;
+  DescriptorStatusList descriptor_statuses;
+};
+
+class RateLimitHeadersTest : public testing::TestWithParam<RateLimitHeadersTestCase> {
+public:
+  static const std::vector<RateLimitHeadersTestCase>& getTestCases() {
+    // clang-format off
```

Comment:
nit: prefer to not turn the formatter off and on. Even if it looks ugly it's better to just let it do its thing.